src/ConcordiumNetSdk/Wallets/WalletAccount.cs

```diff
@@ -0,0 +1,114 @@
+    /// </summary>
+    public readonly AccountAddress AccountAddress;
+
+    /// <summary>
+    /// Internal representation of a signer.
+    /// </summary>
+    private TransactionSigner _signer;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="WalletAccount"/> class.
+    /// </summary>
+    /// <param name="accountAddress">The address of the imported account.</param>
+    /// <param name="signKeys">The signers corresponding to the keys of the imported account.</param>
+    private WalletAccount(
+        AccountAddress accountAddress,
+        Dictionary<AccountCredentialIndex, Dictionary<AccountKeyIndex, ISigner>> signKeys
+    )
+    {
+        AccountAddress = accountAddress;
+        _signer = new TransactionSigner();
+        signKeys
+            .ToList()
+            .ForEach(cred =>
+            {
+                cred.Value
+                    .ToList()
+                    .ForEach(key =>
+                    {
+                        _signer.AddSignerEntry(cred.Key, key.Key, key.Value);
+                    });
+            });
+    }
+
+    /// <summary>
+    /// Try to create a new instance from a wallet data source.
+    /// </summary>
+    /// <param name="walletDataSource">Data source from which to import the wallet.</param>
+    /// <exception cref="WalletDataSourceException"/>
+    public static WalletAccount From(IWalletDataSource walletDataSource)
+    {
+        return new WalletAccount(
+            walletDataSource.TryGetAccountAddress(),
+            walletDataSource.TryGetSignKeys()
+        );
+    }
+
+    /// <summary>
+    /// Try to create a new instance from a JSON string in the genesis wallet key export format.
+    /// </summary>
+    /// <param name="json">JSON string in the genesis wallet key export format.</param>
+    /// <exception cref="JsonException">The specified input is not valid JSON.</exception>
+    /// <exception cref="WalletDataSourceException">Either a field is missing or an index or sign key could not be parsed.</exception>
+    public static WalletAccount FromGenesisWalletExportFormat(string json)
+    {
+        Json.GenesisWalletExportFormat genesisWallet =
+            JsonConvert.DeserializeObject<Json.GenesisWalletExportFormat>(json);
+        return From(genesisWallet);
+    }
+
+    /// <summary>
+    /// Try to create a new instance from a JSON string in the browser wallet key export format.
+    /// </summary>
+    /// <param name="json">JSON string in the browser wallet key export format.</param>
+    /// <exception cref="JsonException">The specified input is not valid JSON.</exception>
+    /// <exception cref="WalletDataSourceException">Either a field is missing or an index or sign key could not be parsed.</exception>
+    public static WalletAccount FromBrowserWalletExportFormat(string json)
+    {
+        Json.BrowserWalletExportFormat genesisWallet =
+            JsonConvert.DeserializeObject<Json.BrowserWalletExportFormat>(json);
```

Comment:
So in the Rust SDK we just have a function where you give it a file or JSON Value and it will figure out what to do.

Here the user has to explicitly choose what they have which makes it more complex to use this, i.e., to make a service which takes any keys.

Is it not possible to do the same as we do in the Rust SDK?

I feel that the interface IWalletDataSource is not really a good solution to this. That should at best be an implementation detail.

Is it not possible to hide this like we do in the Rust SDK? Are there limitations that prevent that?